### PR TITLE
turn on autoescape for flask.templating.render_template_string

### PR DESCRIPTION
**Pull request by alanhamlett**
Previous pull request was #1176

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Could you shorten the commit msg a bit? It's so long that github truncates 
it. Usually the first line should be 51 chars or less (with the 2nd line 
being empty and anything else in the following lines (max 72 chars per line)

**Comment by alanhamlett** [comment]
@ThiefMaster sure, done. What's the eta on this getting merged, I have other pull requests I would like to submit as well.

File: flask/app.py [hunk 1]
```diff
@@ -1460,7 +1460,7 @@ def handle_http_exception(self, e):
         # those unchanged as errors
         if e.code is None:
             return e
-
```
Comment (untitaker) [inline flask/app.py]: I apprechiate this, but could you make it a separate commit?

**Comment by untitaker** [comment]
I'm not sure how, but I think the current behavior should be better documented.

File: flask/app.py (hunk 1 above)
Comment (alanhamlett) [inline flask/app.py]: Sorry, my Vim automatically does that. Fixing it asap.

Comment (alanhamlett) [inline flask/app.py]: @untitaker separated the changes.

**Comment by untitaker** [comment]
In addition to some docs, could you add a changelog and a note in the *Upgrading* document?

**Comment by alanhamlett** [comment]
I've updated the docstring which will show up in [this doc](http://flask.pocoo.org/docs/0.10/api/#flask.render_template_string), but I should regenerate docs or add more info somewhere?

For changelog, should I add to version 0.10.2 or 1.0?

**Comment by untitaker** [comment]
1.0

**Comment by alanhamlett** [comment]
Finished all the changes, everything looks good?

**Comment by ThiefMaster** [comment]
I would squash all the commits (besides the whitespace fixes) into a single one

**Comment by untitaker** [comment]
I've pinged mitsuhiko on IRC if there's a good reason for the current behavior. I assume not and will merge this if he doesn't respond.

**Comment by alanhamlett** [comment]
@ThiefMaster squashed.

**Comment by ThiefMaster** [comment]
thanks! (and oops, looks like i forgot the trailing backtick in the commit message suggestion)

**Comment by untitaker** [comment]
Thanks, rebased and merged!
